src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -722,6 +721,18 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
             )
         else:
             cached_folder = pretrained_model_name_or_path
+            filenames = []
```

Comment:
I think shoud just update the `_identify_model_variants` function using`variant_compatible_siblings`

this PR currently does not fix anything, i.e. it would not have caught the mistake in https://github.com/huggingface/diffusers/pull/9247 

and it is still not able to load variants with shared checkpoints from pipeline level 

i.e.  we should be able to load the fp16 variant in the transformer folder too but it is currently not 
```python
import torch
from diffusers import  AutoPipelineForText2Image
repo = "fal/AuraFlow" # sharded checkpoint with variant
pipe = AutoPipelineForText2Image.from_pretrained(
        repo,
        variant="fp16",
        torch_dtype=torch.float16,
        )
print(pipe.dtype)
```

you get 
```

A mixture of fp16 and non-fp16 filenames will be loaded.
Loaded fp16 filenames:
[vae/diffusion_pytorch_model.fp16.safetensors, text_encoder/model.fp16.safetensors]
Loaded non-fp16 filenames:
[transformer/diffusion_pytorch_model-00002-of-00003.safetensors, transformer/diffusion_pytorch_model-00003-of-00003.safetensors, transformer/diffusion_pytorch_model-00001-of-00003.safetensors
If this behavior is not expected, please check your folder structure.
```

Reply:
cc @DN6 @a-r-r-o-w here too